x-pack/plugins/apm/server/lib/settings/custom_link/get_transaction.ts

```diff
@@ -3,35 +3,53 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-
-import { pick } from 'lodash';
-import {
-  FilterOptions,
-  FILTER_OPTIONS
-} from '../../../../common/custom_link_filter_options';
+import * as t from 'io-ts';
+import { PROCESSOR_EVENT } from '../../../../common/elasticsearch_fieldnames';
 import { Transaction } from '../../../../typings/es_schemas/ui/transaction';
 import { Setup } from '../../helpers/setup_request';
+import { ProcessorEvent } from '../../../../common/processor_event';
+import { FilterOptionsRt } from '../../../routes/settings/custom_link';
 
 export async function getTransaction({
   setup,
   filters = {}
 }: {
   setup: Setup;
-  filters?: FilterOptions;
+  filters?: t.TypeOf<typeof FilterOptionsRt>;
 }) {
   const { client, indices } = setup;
 
-  const esFilters = Object.entries(pick(filters, FILTER_OPTIONS)).map(
-    ([key, value]) => {
-      return { term: { [key]: value } };
-    }
-  );
+  const esFilters = Object.entries(filters)
+    // loops through the filters splitting the value by comma and removing white spaces
+    .map(([key, value]) => {
+      if (value) {
+        return {
+          terms: {
+            [key]: value
+              .split(',')
+              .map(v => v.trim())
+              .filter(v => v)
+          }
+        };
+      }
+    })
+    // removes filters without value
+    .filter(value => value);
```

Comment:
Shouldn't this happen as part of the `convertTo` phase?

Reply:
the `convertTo`, now called `toESFormat`, will receive a CustomLink, but get `get_transaction` receives filters, the toESFormat, will also have the same logic, I can place it inside the same file and call it here too.